### PR TITLE
plex-desktop: enable updateScript & inherit meta

### PR DESCRIPTION
**Pull request by detroyejr**
## Description of changes

Enable automatic updates by setting  passthru.updateScript.

## Things done

Tested with

```
nix-shell maintainers/scripts/update.nix --argstr package plex-desktop
```

And, we get the commit: `plex-desktop: 1.96.0 -> 1.97.2`

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Redhawk18 — APPROVED** [review]

**Comment by Frontear** [comment]
Would the `git add` and `git commit` actions cause issues for @r-ryantm?

**Comment by detroyejr** [comment]
Looking at how other packages do this, my understanding is no this should be fine.

One example is  [git](https://github.com/NixOS/nixpkgs/blob/997332b0816e693d7985138f78757280245f7933/pkgs/applications/version-management/git/update.sh) itself which uses the same `git add` and `git commit` convention in the same script it passes to `passthru.updateScript`. If there is firm documentation that says this isn't correct, send it my way!

**Comment by detroyejr** [comment]
`nixfmt` makes this look like a bigger change, but there are only 2. In addition to  `passthru.updateScript`, I've added `inherit meta` to the buildFhsEnv step just so that search.nixos.org looks nicer.

File: pkgs/by-name/pl/plex-desktop/package.nix [hunk 1]
```diff
@@ -104,48 +105,54 @@ let
...
+  extraInstallCommands = ''
+    mkdir -p $out/share/applications $out/share/icons/hicolor/scalable/apps
+    install -m 444 -D ${plex-desktop}/meta/gui/plex-desktop.desktop $out/share/applications/plex-desktop.desktop
+    substituteInPlace $out/share/applications/plex-desktop.desktop \
+      --replace-fail \
+      'Icon=''${SNAP}/meta/gui/icon.png' \
+      'Icon=${plex-desktop}/meta/gui/icon.png' \
+      --replace-fail \
+      'Exec=plex-desktop' \
+      'Exec=plex-desktop-${version}'
```
Comment (SuperSandro2000) [inline pkgs/by-name/pl/plex-desktop/package.nix:137]:
```suggestion
      --replace-fail 'Icon=''${SNAP}/meta/gui/icon.png' 'Icon=${plex-desktop}/meta/gui/icon.png' \
      --replace-fail 'Exec=plex-desktop' 'Exec=plex-desktop-${version}'
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
The formatting must be in an extra commit.

**Comment by detroyejr** [comment]
Should be resolved.

Thanks for the review!

**Pull request closed without merging**